rest-api/core-rest/src/main/java/uk/gov/gchq/gaffer/rest/service/v2/IOperationServiceV2.java

```diff
@@ -0,0 +1,121 @@
+import static uk.gov.gchq.gaffer.rest.ServiceConstants.FORBIDDEN;
+import static uk.gov.gchq.gaffer.rest.ServiceConstants.INTERNAL_SERVER_ERROR;
+import static uk.gov.gchq.gaffer.rest.ServiceConstants.OK;
+import static uk.gov.gchq.gaffer.rest.ServiceConstants.OPERATION_NOT_FOUND;
+import static uk.gov.gchq.gaffer.rest.ServiceConstants.OPERATION_NOT_IMPLEMENTED;
+
+/**
+ * An <code>IOperationServiceV2</code> has methods to execute {@link uk.gov.gchq.gaffer.operation.Operation}s on the
+ * {@link uk.gov.gchq.gaffer.graph.Graph}.
+ */
+@Path("/graph/operations")
+@Api(value = "operations")
+@Produces(APPLICATION_JSON)
+@Consumes(APPLICATION_JSON)
+public interface IOperationServiceV2 {
+
+    @GET
+    @ApiOperation(value = "Gets all operations supported by the store.", response = String.class, responseContainer = "list", produces = APPLICATION_JSON)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = OK)})
+    Response getOperations();
+
+    @POST
+    @ApiOperation(value = "Performs the given operation on the graph", response = Object.class, produces = APPLICATION_JSON)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = OK),
+            @ApiResponse(code = 400, message = BAD_REQUEST),
+            @ApiResponse(code = 403, message = FORBIDDEN),
+            @ApiResponse(code = 500, message = INTERNAL_SERVER_ERROR),
+            @ApiResponse(code = 501, message = OPERATION_NOT_IMPLEMENTED)})
+    Response execute(final Operation operation);
+
+    @POST
+    @Path("/chunked")
+    @ApiOperation(value = "Performs the given operation on the graph, returned chunked output. NOTE - does not work in Swagger.", response = Object.class, produces = APPLICATION_JSON)
+    @ApiResponses(value = {@ApiResponse(code = 202, message = OK),
+            @ApiResponse(code = 400, message = BAD_REQUEST),
+            @ApiResponse(code = 403, message = FORBIDDEN),
+            @ApiResponse(code = 500, message = INTERNAL_SERVER_ERROR),
+            @ApiResponse(code = 501, message = OPERATION_NOT_IMPLEMENTED)})
+    ChunkedOutput<String> executeChunked(final Operation operation);
+
+    @SuppressFBWarnings
+    ChunkedOutput<String> executeChunkedChain(final OperationChain opChain);
+
+    @GET
+    @Path("/{className}")
+    @ApiOperation(value = "Gets details about the specified operation class.", produces = APPLICATION_JSON)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = OK),
+            @ApiResponse(code = 403, message = FORBIDDEN),
+            @ApiResponse(code = 404, message = OPERATION_NOT_FOUND),
+            @ApiResponse(code = 500, message = INTERNAL_SERVER_ERROR)})
+    Response operationDetails(@ApiParam(value = "the fully qualified class name") @PathParam("className") final String className) throws InstantiationException, IllegalAccessException;
+
+    @POST
+    @Path("/{className}")
+    @ApiOperation(value = "Performs the given operation on the graph", response = Object.class, produces = APPLICATION_JSON)
```

Comment:
I don't think we need this end point. It just duplicates the POST at /graph/operations with an extra path param that isn't actually used.